tests/test_utils_typing.py

```diff
@@ -47,3 +53,70 @@ def test_is_jsonable_failure(data):
     assert not is_jsonable(data)
     with pytest.raises((TypeError, ValueError)):
         json.dumps(data)
+
+
+@pytest.mark.parametrize(
+    "type_, is_optional",
+    [
+        (Optional[int], True),
+        (Union[None, int], True),
+        (Union[int, None], True),
+        (Optional[CustomType], True),
+        (Union[None, CustomType], True),
+        (Union[CustomType, None], True),
+        (int, False),
+        (None, False),
+        (Union[int, float, None], False),
+        (Union[Union[int | float], None], False),
+        (Optional[int | float], False),
+    ],
+)
+def test_is_simple_optional_type(type_: Type, is_optional: bool):
+    assert is_simple_optional_type(type_) == is_optional
+
+
+@pytest.mark.skipif(sys.version_info < (3, 10), reason="requires python3.10 or higher")
```

Comment:
I was about to put a comment for those but you already tested them. Thank you! :heart: